pulp_python/app/utils.py

```diff
@@ -0,0 +1,64 @@
+import json
+
+
+def parse_project_metadata(project):
+    """
+    Create a dictionary of python project metadata
+
+    Args:
+        project (dict): of metadata relevant to the entire Python project
+
+    Returns:
+        dictionary: of python project metadata
+    """
+
+    package = {}
+    package['name'] = project.get('name') or ""
+    package['metadata_version'] = project.get('metadata_version') or ""
+    package['summary'] = project.get('summary') or ""
+    package['description'] = project.get('description') or ""
+    package['keywords'] = project.get('keywords') or []
+    package['home_page'] = project.get('home_page') or ""
+    package['download_url'] = project.get('download_url') or ""
+    package['author'] = project.get('author') or ""
+    package['author_email'] = project.get('author_email') or ""
+    package['maintainer'] = project.get('maintainer') or ""
+    package['maintainer_email'] = project.get('maintainer_email') or ""
+    package['license'] = project.get('license') or ""
+    package['requires_python'] = project.get('requires_python') or ""
+    package['project_url'] = project.get('project_url') or ""
+    package['platform'] = project.get('platform') or ""
+    package['supported_platform'] = project.get('supported_platform') or ""
+    package['requires_dist'] = json.dumps(project.get('requires_dist', []))
+    package['provides_dist'] = json.dumps(project.get('provides_dist', []))
+    package['obsoletes_dist'] = json.dumps(project.get('obsoletes_dist', []))
+    package['requires_external'] = json.dumps(project.get('requires_external', []))
+
+    return package
+
+
+def parse_metadata(project, version, distribution):
+    """
+    Create a dictionary of metadata needed to create a PythonContentUnit from
+    the project, version, and distribution metadata
+
+    Args:
+        project (dict): of metadata relevant to the entire Python project
+        version (string): version of distribution
+        distribution (dict): of metadata of a single Python distribution
+
+    Returns:
+        dictionary: of useful python metadata
+    """
+
+    package = {}
+
+    package['filename'] = distribution.get('filename') or ""
+    package['packagetype'] = distribution.get('packagetype') or ""
+    package['version'] = version
+    package['url'] = distribution.get('url') or ""
+    package['md5_digest'] = distribution.get('md5_digest') or ""
```

Comment:
could you change this to use sha256?

https://github.com/pulp/pulp_python/pull/154

Reply:
Is sha256_digest (`distribution['digests']['sha256']`) guaranteed to be present in the json API?

Reply:
according to this, it should, but the key is nested under "digests" instead of being at the top level like ```md5_digest``` https://warehouse.pypa.io/api-reference/json/